gridfs_bucket.go

```diff
@@ -115,6 +99,13 @@ func (b *gridfsBucket) Exists(ctx context.Context, key string) (bool, error) {
 
 func (b *gridfsBucket) Join(elems ...string) string { return consistentJoin(elems) }
 
+// bucket returns a new GridFS bucket configured with the context timeout, if
+// it exists. This function is called by each operation that needs to read or
+// write from the bucket to avoid read and write deadline conflicts—custom
+// deadlines cannot be set on a GridFS bucket instance concurrently with other
+// read or write operations that also require a custom deadlines. Setting the
```

Comment:
nit: maybe phrase the last sentence as `This function sets the read and write deadlines to allow it to respect the context timeouts passed in by the caller`, since the phrasing seems to assume you already know it sets the read/write deadlines.